packages/@uppy/core/src/Uploader.js

```diff
@@ -0,0 +1,144 @@
+const { nanoid } = require('nanoid/non-secure')
+
+class Uploader {
+  preProcessors = new Set()
+
+  uploaders = new Set()
```

Comment:
I'm feel like exposing these as public properties is not so clean, if they are the concern of the Uploader class

Reply:
This is impossible to make private as they can be updated by any plugin. The principle here is logic that changes together, stays together. Therefor I think it makes sense to have it there, publicly.

Reply:
Uploader is still private in Uppy, and therefor these are too, but these must be accessible from the uppy class. So they can't be private inside Uploader.

```js
  #uploader = new Uploader()
```

Reply:
as far as I can see only `postProcessors` is used from Uppy in `this.#uploader.postProcessors.size > 0`. I think it would be better to expose a public method called `hasPostProcessors = () => this.#postProcessors.size > 0`